skbio/io/registry.py

```diff
@@ -558,11 +558,28 @@ def _init_reader(self, file, fmt, into, verify, kwargs, io_kwargs):
 
         reader = self.get_reader(fmt, into)
         if reader is None:
+            possible_readers = self._get_possible_readers(fmt)
+            if possible_readers:
+                possible_intos = []
+                for reader in possible_readers:
+                        possible_intos.append(reader.__name__)
+                message = """Possible values for `into` include: %s
+                          """ % ', '.join(possible_intos)
+            else:
+                message = ''
             raise UnrecognizedFormatError(
-                "Cannot read %r from %r, no %s reader found." %
-                (fmt, file, into.__name__ if into else 'generator'))
+                "Cannot read %r from %r, no %s reader found. `into` also"
+                " not found. %s" %
+                (fmt, file, into.__name__ if into else 'generator',
```

Comment:
This ternary suggests that sometimes `into` is set, so the error will be incorrect sometimes.